src/azfilenfs-watchdog

```diff
@@ -0,0 +1,222 @@
+        # any locking that we dependent on the old file.
+        #
+        out=$(sed "\%^${1}$%d" $AZ_FILES_MOUNTMAP)
+        ret=$?
+        if [ $ret -eq 0 ]; then
+            #
+            # If this echo fails then AZ_FILES_MOUNTMAP could be truncated.
+            #
+            echo "$out" > $AZ_FILES_MOUNTMAP
+            ret=$?
+            out=
+            if [ $ret -ne 0 ]; then
+                eecho "*** [FATAL] AZ_FILES_MOUNTMAP may be in inconsistent state, contact Microsoft support ***"
+            fi
+        fi
+
+        if [ $ret -ne 0 ]; then
+            chattr -f +i $AZ_FILES_MOUNTMAP
+            eecho "[$1] failed to remove from ${AZ_FILES_MOUNTMAP}!"
+            return 1
+        fi
+        chattr -f +i $AZ_FILES_MOUNTMAP
+
+	# Return the mtime after our mods.
+        echo $(stat -c%Y $AZ_FILES_MOUNTMAP)
+    ) 999<$AZ_FILES_MOUNTMAP
+}
+
+
+vecho "Starting azfilenfs-watchdog..."
+
+if [ ! -f $AZ_FILES_MOUNTMAP ]; then
+	touch $AZ_FILES_MOUNTMAP
+	if [ $? -ne 0 ]; then
+		eecho "[FATAL] Not able to create '${AZ_FILES_MOUNTMAP}'!"
+		exit 1
+	fi
+fi
+
+if ! chattr -f +i $AZ_FILES_MOUNTMAP; then
+    wecho "chattr does not work for ${AZ_FILES_MOUNTMAP}!"
+fi
+
+#
+# Watchdog for monitoring unmounts and stunnel processes
+#
+while :; do
+    sleep $MONITOR_INTERVAL_SECS
+
+    epoch_now=$(date +%s)
+
+    #
+    # Go over all lines in $AZ_FILES_MOUNTMAP and check them for two things:
+    # 1. Is that entry still in use by at least one aznfs mount, if not remove the entry.
+    # 2. Is stunnel process running?
+    #
+    # We store the mtime of AZ_FILES_MOUNTMAP while inside the lock so that if any mount helper process
+    # updates it after this we will skip modification for sake of safety. We will come to it
+    # in the next iteration when it's safer.
+    #
+    exec {fd}<$AZ_FILES_MOUNTMAP
+    flock -e $fd
+    mtime_mountmap=$(stat -c%Y $AZ_FILES_MOUNTMAP)
+    IFS=$'\n' lines=$(cat $AZ_FILES_MOUNTMAP)
+    flock -u $fd
+    exec {fd}<&-
+
+    #
+    # findmnt must be done after reading AZ_FILES_MOUNTMAP so that if we come accross a
+    # AZ_FILES_MOUNTMAP entry whose all nfs file shares are unmounted, we know
+    # for sure that it's not in use by any mount and can be removed.
+    #
+    findmnt=$(findmnt | grep 'nfs4\|127.0.0.1' 2>&1)
+
+    #
+    # For no matching mounts also, findmnt exits with a failure return, so check
+    # for both exit status and non-empty error o/p.
+    #
+    if [ $? -ne 0 -a -n "$findmnt" ]; then
+        eecho "${findmnt}."
+        eecho "[FATAL] findmnt failed unexpectedly!"
+        eecho "[FATAL] Azfilenfs-watchdog service is exiting, will not monitor Azure NFS file shares."
+        eecho "[FATAL] Please contact Microsoft support before using any NFS File shares."
+        # This usually indicates some non-transient issue, bail out.
+        exit 1
+    fi
+
+    for line in $lines; do
```

Comment:
lines were not supposed to be greater than certain predefined numbe?

Reply:
We are not enforcing that limit for nfsv4 for now. But we may need to enforce it in future when we will support account migration.